src/search-manager/data/api.ts

```diff
@@ -551,3 +556,24 @@ export async function fetchDocumentById({ client, indexName, id } : {
   const doc = await client.index(indexName).getDocument(id);
   return formatSearchHit(doc);
 }
+
+/**
+ * Fetch a content hit by usage key
+ */
+export const fetchContentByUsageKey = async (
```

Comment:
Wouldn't adding the collection data to the `getLibraryBlockMetadata` return be better?
It will make things simpler on this side (we wouldn't need another context only for this).

Reply:
@rpenido Yes, updated the API include collections data along with component.